src/keytypes/trusted.rs

```diff
@@ -211,22 +194,20 @@ pub enum Payload {
 
 impl KeyPayload for Payload {
     fn payload(&self) -> Cow<[u8]> {
-        let (command, blob, options) = match *self {
+        match self {
             Payload::New {
-                ref keylen,
-                ref options,
-            } => ("new", format!("{}", keylen), options),
+                keylen,
+                options,
+            } => format!("new {}{}", keylen, options),
             Payload::Load {
-                ref blob,
-                ref options,
-            } => ("load", AsciiHex::convert(&blob), options),
+                blob,
+                options,
+            } => format!("load {:x}{}", ByteBuf(blob), options),
             Payload::Update {
-                ref options,
-            } => ("update", String::new(), options),
-        };
-
-        format!("{} {} {}", command, blob, options.payload_string())
-            .bytes()
-            .collect()
+                options,
+            } => format!("update {}", options),
```

Comment:
Won't this leave a double space after `update`?

Reply:
Correct, this was the behavior before, so I left it in, but it was probably not intentional, so I've now fixed it.